### DISCUSSION TITLE
Error: export 'isSupported' (imported as 'isSupported') was not found in 'firebase/remote-config'

### DISCUSSION THREAD

**Discussion by Synkretik** [angular/angularfire#3008] [Q&A]
Hi,

I installed the angularfire master on my localhost.
```bash
> yarn install  // in root
> yarn build // in root
> yarn install // in root/samples/.emulator/functions
> yarn build// in root/samples/.emulator/functions
> yarn install // in root/samples/compat & modular
> yarn build // in root/samples/compat & modular
> yarn start // in root/samples/compat & modular
```

At which point, I get:
```
NOTICE: Hot Module Replacement (HMR) is enabled for the dev server.
See https://webpack.js.org/guides/hot-module-replacement for information on working with HMR for Webpack.
√ Browser application bundle generation complete.

Initial Chunk Files                                             | Names         |      Size
vendor.js                                                       | vendor        |   5.27 MB
polyfills.js                                                    | polyfills     |   1.09 MB
styles.css, styles.js                                           | styles        | 392.97 kB
main.js                                                         | main          |  79.72 kB
runtime.js                                                      | runtime       |  50.38 kB

                                                                | Initial Total |   6.88 MB

Lazy Chunk Files                                                | Names         |      Size
node_modules_firebase_compat_performance_dist_index_esm_js.js   | -             |  64.69 kB
node_modules_firebase_compat_messaging_dist_index_esm_js.js     | -             |  61.90 kB
node_modules_firebase_compat_functions_dist_index_esm_js.js     | -             |  30.67 kB
src_app_protected-lazy_protected-lazy_module_ts.js              | -             |   8.29 kB
node_modules_firebase_compat_analytics_dist_index_esm_js.js     | -             |   7.16 kB
node_modules_firebase_compat_remote-config_dist_index_esm_js.js | -             |   5.89 kB

Build at: 2021-10-09T23:52:57.679Z - Hash: 94b3db19614f38d6a7d0 - Time: 9019ms

./node_modules/@angular/fire/fesm2015/angular-fire-compat-remote-config.js:112:99-110 - Error: export 'isSupported' (imported as 'isSupported') was not found in 'firebase/remote-config' (possible exports: activate, ensureInitialized, fetchAndActivate, fetchConfig, g
etAll, getBoolean, getNumber, getRemoteConfig, getString, getValue, setLogLevel)

./node_modules/@angular/fire/fesm2015/angular-fire.js:18:104-117 - Error: export 'isSupported' (imported as 'isSupported$2') was not found in 'firebase/remote-config' (possible exports: activate, ensureInitialized, fetchAndActivate, fetchConfig, getAll, getBoolean,
getNumber, getRemoteConfig, getString, getValue, setLogLevel)



** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

```

opening `http://localhost:4200/ `  yields  `Cannot GET /`

Any suggestions on how to run the samples would be appreciated.

**Comment by jamesdaniels** [comment]
The latest AngularFire requires Firebase JS SDK v9.1+, update the dependency and it should start working.

**Reply by loonylabs-dev** [reply]
Hi,
I have the same issue and it seems that my versions are fine. I have this issue since today. When I have installed ngx-charts with npm I have done 'npm audit fix' afterwars when I saw that there are some vulnerabilities. Here my versions from package.json

` "dependencies": {
    "@angular/animations": "~13.1.0",
    "@angular/common": "~13.1.0",
    "@angular/compiler": "~13.1.0",
    "@angular/core": "~13.1.0",
    "@angular/fire": "^7.2.1",
    "@angular/forms": "~13.1.0",
    "@angular/platform-browser": "~13.1.0",
    "@angular/platform-browser-dynamic": "~13.1.0",
    "@angular/router": "~13.1.0",
    "@swimlane/ngx-charts": "^20.0.1",
    "bootstrap": "^5.1.3",
    "firebase": "^9.6.6",
    "rxjs": "~7.4.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  }`
  
  specific error message:
  ./node_modules/@angular/fire/fesm2015/angular-fire.js:17:98-111 - Error: export 'isSupported' (imported as 'isSupported$1') was not found in 'firebase/messaging' (module has no exports)ing' (module has no exports)
  
  
  this issue is happening with command 'ng serve --open' (with command 'ng serve' I get not issue)
  
  I have done uninstall 
  * npm uninstall @angular/fire
  * npm uninstall firebase
  * npm uninstall fire

and afterwards installed with @latest. I tried also some specific versions but I am not able to get it running. Any help is much appreciated

**Comment by iztokb** [comment]
I have the same problem when upgrading to angular v14, 
"@angular/fire": "7.4.1", 
"firebase": "9.8.3",

./node_modules/@angular/fire/fesm2015/angular-fire.js:16:98-109 - Error: export 'isSupported' (imported as 'isSupported') was not found in 'firebase/analytics' (module has no exports)

**Reply by Villanuevand** [reply]
Hello @iztokb , 
I recommend trying to delete the `node_modules` folder and make a clean installation. 
If you take a look to `firebase.ts` file [src/analytics](https://github.com/angular/angularfire/blob/3ae6ce5dff0da50fea456d9c926d360cc50f05d3/src/analytics/firebase.ts#L15)  line 15, the build tool is exporting the `isSupported` property properly. 

I would like how this issue ends, let me know if you have any update.

**Comment by megazordy** [comment]
Hi,

I was successful with this problem after removing the 'node_modules' folder, and running the 'npm install' and 'npm update --force' commands.

package.json:
{
  "firebase": "^9.8.4",
  "firebase-admin": "^10.0.1",
   "firebase-functions": "^3.16.0",
  "@angular/fire": "^7.2.0",
}

**Comment by alexfung888** [comment]
I recently got the same problem with angular@16, firebase@10.4.0, @angular/fire@7.6.1

Deleting `nodes_module` did not help.
I also deleted the` yarn.lock` and then `npm start` works again.

**Reply by rgant** [reply]
Angular Fire does not support firebase@10. https://github.com/angular/angularfire#angular-and-firebase-versions

**Reply by alexfung888** [reply]
Only documentation says no. Has been working fine.

[package.json](https://github.com/angular/angularfire/blob/5793d6f3a15aa61660af75ab361c534cde9b73df/package.json#L58C7-L58C37)

since [this commit](https://github.com/angular/angularfire/commit/5793d6f3a15aa61660af75ab361c534cde9b73df)

**Reply by rgant** [reply]
But you said you have installed this commit: https://github.com/angular/angularfire/blob/b4c3da8db8d07ad4bbb34d81f835862658cc441c/package.json#L57

**Comment by itsbrianburton** [comment]
For anyone else trying to get AngularFire 7.6.1 to play nicely with Firebase, you need to install 9.8.1 specifically:

```
npm uninstall firebase --save
npm install firebase@9.8.1 --save
```

**Do not copy the semver from AF's package.json, which looks like this:**

`npm install firebase@^9.8.0 --save`

The above will install the latest version 9, currently 9.23.0, which is not compatible with AF 7.6.1.  Firebase 9.8.0 contains typings issues that will prevent you from compiling.  Later 9.x versions of Firebase may work, but I can confirm that 9.8.1 is compatible.
